src/dialogxml/dialogs/dialog.cpp

```diff
@@ -771,13 +771,28 @@ bool cDialog::setFocus(cTextField* newFocus, bool force) {
 		currentFocus = "";
 		return true;
 	}
-	auto iter = find_if(controls.begin(), controls.end(), [newFocus](std::pair<const std::string, cControl*> p){
-		return p.second == newFocus;
-	});
-	if(iter == controls.end()) return false;
-	if(!force && !newFocus->triggerFocusHandler(*this, iter->first, false)) return false;
-	currentFocus = iter->first;
-	return true;
+	for(auto iter = controls.begin(); iter != controls.end(); iter++) {
+		if(iter->second == newFocus) {
+			if (!force && !newFocus->triggerFocusHandler(*this, iter->first, false)) return false;
+			currentFocus = iter->first;
+			return true;
+		}
+		if(iter->second->isContainer()) {
+			cContainer* tmp = dynamic_cast<cContainer*>(iter->second);
+			std::string foundKey = "";
+			tmp->forEach([&](std::string key, cControl& child) {
```

Comment:
Generally I prefer to explicitly list what needs to be captured, rather than using `[&]` or `[=]` to capture everything.